Compiling area cleaned/updated

SaturnusDJ opened the pull request:
- Moved icons dir into build dir, updated references to it
- Moved dmgbg.png into build dir, updated references to it
- Moved dmg.json into build dir, updated references/metadata in it and to it
- Changed branding in build.json
- Added buildDir in build.json so we have control over the output
  location
- Added cacheDir in build.json so cache is in a better location and not
  in root anymore
- Updated the paths to the changes above
- The first line of installer-script.iss opened with strange signs,
  fixed
- Changed branding and metadata in installer-script.iss
- Updated installer-script.iss with new paths
- Made paths in installer-script.iss relative, so it works for everybody
  using the repo
- Updated gitignore
- Confirmed the compiling proces is still working (None related bug:
  https://github.com/tinderjs/tinder-desktop/issues/5#issuecomment-207819152)

SaturnusDJ committed:
Compiling area cleaned/updated

- Moved icons dir into build dir, updated references in it and to it
- Moved dmgbg.png into build dir, updated references in it and to it
- Moved dmg.json into build dir, updated references in it and to it

- Changed branding in build.json
- Added buildDir in build.json so we have control over the output
location
- Added cacheDir in build.json so cache is in a better location and not
in root anymore
- Updated the paths to the changes above

- The first line of installer-script.iss opened with strange signs,
fixed
- Changed branding and metadata in installer-script.iss
- Updated installer-script.iss with new paths
- Made paths in installer-script.iss relative, so it works for everybody
using the repo

- Updated gitignore

- Confirmed the compiling proces is still working (None related bug:
https://github.com/tinderjs/tinder-desktop/issues/5#issuecomment-207819152)

SaturnusDJ committed:
Update build.js

stuwil commented on build.js:
```diff
@@ -36,7 +39,7 @@ function createNW() {
   var archive = archiver('zip');
 
-  var output = fs.createWriteStream('./build/' + appName + '/tinder-' + appPkg.version + '.nw');
+  var output = fs.createWriteStream(buildDir + '/tinder-' + appPkg.version + '.nw');
```
Can we change this to tinder-desktop-{{version}}.nw?

SaturnusDJ replied:
Done in e02752d

SaturnusDJ commented on build.js:
```diff
@@ -3,16 +3,19 @@ console.log('Before running, make sure versions are updated in both package.json
 var NwBuilder = require('node-webkit-builder');
 var appPkg = require('./desktop-app/package.json');
-var appName = 'Tinder⁺⁺';
+var appName = 'Tinder Desktop';
```
`var appName = 'Tinder Desktop';` is probably internal too so can be tinder-desktop, but then the paths in installer-script.iss containing 'Tinder Desktop' need to be renamed accordingly.

stuwil replied:
Fine with that... we really shouldn't have spaces in our build path.

SaturnusDJ committed:
Using tinder-desktop as path

Tinder Desktop is the name of the program directory after installing.
Inside is tinder-desktop.exe
All shortcuts to it (quick launch, start menu, desktop) are named Tinder
Desktop.

SaturnusDJ committed:
Updated nwjs version used for compiling to 0.12.3

Previous was 0.11.6

SaturnusDJ committed:
Rename to tinder-desktop

Confirmed to be working

SaturnusDJ committed:
Final edit so the running exe has the correct process name

stuwil commented on build/installer-script.iss:
```diff
@@ -1,11 +1,11 @@
-#define MyAppURL "http://www.tinderplusplus.com/"
-#define MyAppExeName "Tinder⁺⁺.exe"
+#define MyAppPublisher "TinderJS, Github Organization"
+#define MyAppURL "https://tinderjs.com"
```
Use https://github.com/tinderjs/tinder-desktop

SaturnusDJ committed:
URL change to Github
